utils/post.js

```diff
@@ -36,14 +36,14 @@ const UserFollowUserFunction = require('../databases/functions/userFollowUser');
 const filterAllTopics = (text, topics = []) => {
   const topicsFromText = text.match(/#([A-Z0-9-_]+)\b/gi) || [];
   const topicsFromTextWithoutHashtag = topicsFromText.reduce((acc, next) => {
-    acc.push(next.slice(1));
+    acc.push(next.replace(/#/g, ''));
     return acc;
   }, []);
 
   let unique_topics = [
     ...new Set([
       ...topicsFromTextWithoutHashtag.map((topic) => topic.toLowerCase().trim()),
-      ...topics.map((topic) => topic.toLowerCase().trim())
+      ...topics.map((topic) => topic.replace(/#/g, '').toLowerCase().trim())
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for removing '#' characters from topics and making them unique is correct. However, it can be optimized by using a single `Set` operation instead of two separate operations (creating an array and then converting it to a set). This will improve the performance of your code.

```diff
- let unique_topics = [
-   ...new Set([
-     ...topicsFromTextWithoutHashtag.map((topic) => topic.toLowerCase().trim()),
-     ...topics.map((topic) => topic.replace(/#/g, '').toLowerCase().trim())
-   ])
- ];
+ let unique_topics = Array.from(new Set([
+   ...topicsFromTextWithoutHashtag,
+   ...topics
+ ].map((topic) => topic.replace(/#/g, '').toLowerCase().trim())));
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->